sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JdbcRelationProvider.scala

```diff
@@ -39,7 +39,7 @@ class JdbcRelationProvider extends CreatableRelationProvider
       null
     } else {
       JDBCPartitioningInfo(
-        partitionColumn, lowerBound.toLong, upperBound.toLong, numPartitions.toInt)
+        partitionColumn, lowerBound.toLong, upperBound.toLong, numPartitions.get)
```

Comment:
If you are doing this, how about changing partitionColumn, lowerBound, upperBound to Option types as well?